pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+let timeout = null
+
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
+    setDataProviderUrl(newUrl)
+  }
+
+  useDebouncedEffect(() => {
+    if (dataProviderUrl === '') return
+
+    setIsLoading(true)
+
+    // TODO: Call API Here
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      if (dataProviderUrl === 'success.url')
+        setDiscoveredUrls([{ url: dataProviderUrl, isNew: true }])
+      else if (dataProviderUrl === 'duplicate.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }])
+      else if (dataProviderUrl === 'multiple-duplicates.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }, { url: dataProviderUrl }])
+      else setDiscoveredUrls([])
+
+      setIsLoading(false)
+    }, 1000)
+  }, [dataProviderUrl])
+
+  useEffect(() => {
+    setDataProviderUrl(defaultDataProviderUrl)
+  }, [defaultDataProviderUrl])
+
+  return [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ]
```

Comment:
I am afraid this interface is not very useful. With a tuple (2 items) it's obvious but here are too many options to make a mistake.

I would rather return a kind of 'store' object.

Reply:
that's true, thanks